File: edison/src/main/java/com/prefanatic/edisoniot/SocketServer.java
package com.prefanatic.edisoniot;

import org.java_websocket.WebSocket;
import org.java_websocket.handshake.ClientHandshake;
import org.java_websocket.server.WebSocketServer;

import java.net.InetSocketAddress;

/**
 * Created by codygoldberg on 12/26/16.
 */

public class SocketServer extends WebSocketServer {

    private WebSocket mSocket;

    public SocketServer(InetSocketAddress address) {
        super(address);
    }

    @Override
    public void onOpen(WebSocket conn, ClientHandshake handshake) {
        mSocket = conn;

        mSocket.send("Hello Client!");
    }

    @Override
    public void onClose(WebSocket conn, int code, String reason, boolean remote) {

    }

    @Override
    public void onMessage(WebSocket conn, String message) {

    }

    @Override
    public void onError(WebSocket conn, Exception ex) {

    }
}

File: edison/src/main/java/com/prefanatic/edisoniot/SocketServer.kt
package com.prefanatic.edisoniot

import android.util.Log
import com.squareup.moshi.Moshi

import org.java_websocket.WebSocket
import org.java_websocket.handshake.ClientHandshake
import org.java_websocket.server.WebSocketServer

import java.net.InetSocketAddress
import java.nio.ByteBuffer

import rx.Observable
import rx.subjects.PublishSubject

/**
 * Created by codygoldberg on 12/26/16.
 */

class SocketServer(address: InetSocketAddress) : WebSocketServer(address) {
    private var mSocket: WebSocket? = null
    private val delegate = RxSocketDelegate<WebSocket>()

    override fun onOpen(conn: WebSocket, handshake: ClientHandshake) {
        mSocket = conn

        delegate.onOpened(conn)
    }

    override fun onClose(conn: WebSocket, code: Int, reason: String, remote: Boolean) {
        delegate.onClosed()
    }

    override fun onMessage(conn: WebSocket, message: String) {
        Log.d(TAG, "onMessage(String): " + message)

        delegate.onMessage(message)
    }

    override fun onMessage(conn: WebSocket?, message: ByteBuffer?) {
        Log.d(TAG, "onMessage(ByteBuffer): " + String(message!!.array()))

        //byteMessageSubject.onNext(message)
    }

    override fun onError(conn: WebSocket, ex: Exception) {
        delegate.onError(ex)
    }

    fun asObservable(): Observable<Message> {
        return delegate.messageSubject.asObservable()
    }

    companion object {
        private val TAG = "SocketServer"
    }
}